client/v2/client.go

```diff
@@ -413,16 +414,45 @@ func (uc *udpclient) Write(bp BatchPoints) error {
 	for _, p := range bp.Points() {
 		pointstring := p.pt.RoundedString(d) + "\n"
 
-		// Write and reset the buffer if we reach the max size
 		if b.Len()+len(pointstring) >= uc.payloadSize {
 			if _, err := uc.conn.Write(b.Bytes()); err != nil {
 				return err
 			}
 			b.Reset()
 		}
 
-		if _, err := b.WriteString(pointstring); err != nil {
-			return err
+		if b.Len()+len(pointstring) <= uc.payloadSize {
+			if _, err := b.WriteString(pointstring); err != nil {
+				return err
+			}
+		} else {
+			if p.Time().IsZero() {
+				if len(pointstring) > MaxPointSize {
+					return fmt.Errorf("point size is greater than %dB", MaxPointSize)
+				}
+				if _, err := b.WriteString(pointstring); err != nil {
+					return err
+				}
+			} else {
+				points, err := p.pt.SplitN(uc.payloadSize - 2) // -2 because of `+ "\n"`
+				if err != nil {
+					return err
```

Comment:
SplitN splits a point into multiple points and ensures that
the string length of any returned point is no larger than the input size,
**except points which consists from one field, they can be larger than the input size.**

Or should it produce an error in this case?